components/Request/Request.jsx

```diff
@@ -8,6 +8,7 @@ import { REDUCER_KEY } from '../../reducers';
 
 const Request = ({ error, matchId, loading, progress, dispatchRequest, dispatchMatchId }) => {
   function submit() {
+    document.getElementById('match_id').value = '';
```

Comment:
I think you should dispatchMatchId here.  React discourages direct mutation of DOM state.